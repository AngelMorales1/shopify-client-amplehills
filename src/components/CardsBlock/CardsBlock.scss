@import 'settings', 'src/styles/utils';

.CardsBlock {
  flex-direction: column;

  @media ($medium-up) {
    flex-direction: row;
    justify-content: center;
  }

  &__card {
    height: spacing(85);
  }

  &__card--dark-blue {
    @extend .bg-dark-bay-blue;
  }

  &__card--yellow {
    @extend .bg-butter-yellow;
  }

  &__card--peach {
    @extend .bg-sea-peach;
  }

  &__card--navy {
    @extend .bg-navy;
  }

  &__card--blue {
    @extend .bg-bay-blue;
  }

  &__card--purple {
    @extend .bg-purple;
  }

  &__card-image {
    height: 60%;
    width: 60%;
    object-fit: contain;

    @media ($medium-up) {
      width: 40%;
    }
  }
}
