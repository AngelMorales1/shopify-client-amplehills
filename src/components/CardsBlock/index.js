import React from 'react';
import PropTypes from 'prop-types';
import get from 'utils/get';
import contentfulImgUtil from 'utils/contentfulImgUtil';
import cx from 'classnames';

import styles from './CardsBlock.scss';
import { Button, Image } from 'components/base';

const CardsBlock = ({ cardsBlock }) => {
  const cardBlock1 = get(cardsBlock, 'cardBlock1', null);
  const cardBlock2 = get(cardsBlock, 'cardBlock2', null);

  return (
    <div
      className={cx(
        styles['CardsBlock'],
        'w100 p3 mt4 flex content-width mx-auto'
      )}
    >
      {cardBlock1 ? (
        <div className="col-12 md-col-6">
          <Button
            to={cardBlock1.link}
            className="text-white"
            variant="style-none"
          >
            <div
              className={cx(
                styles['CardsBlock__card'],
                styles[`CardsBlock__card--${cardBlock1.color}`],
                'flex flex-column items-center justify-between p3 m1'
              )}
            >
              <Image
                className={cx(styles['CardsBlock__card-image'], 'col-4 mt2')}
                src={cardBlock1.image}
              />
              <p className="py2">{cardBlock1.text}</p>
            </div>
          </Button>
        </div>
      ) : null}
      {cardBlock2 ? (
        <div className="col-12 md-col-6">
          <Button
            to={cardBlock2.link}
            className="text-white"
            variant="style-none"
          >
            <div
              className={cx(
                styles['CardsBlock__card'],
                styles[`CardsBlock__card--${cardBlock2.color}`],
                'flex flex-column items-center justify-between p3 m1'
              )}
            >
              <Image
                className={cx(styles['CardsBlock__card-image'], 'col-4 mt2')}
                src={cardBlock2.image}
              />
              <p className="py2">{cardBlock2.text}</p>
            </div>
          </Button>
        </div>
      ) : null}
    </div>
  );
};

CardsBlock.propTypes = {};

CardsBlock.defaultProps = {};

export default CardsBlock;
