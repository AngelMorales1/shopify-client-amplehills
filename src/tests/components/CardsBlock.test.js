import React from 'react';
import { shallow } from 'enzyme';

import CardsBlock from 'components/CardsBlock';

it('renders without data', () => {
  const component = shallow(<CardsBlock />);

  expect(component).toMatchSnapshot();
});

it('renders with data', () => {
  const component = shallow(
    <CardsBlock
      cardBlock1={{
        color: 'foo',
        image: 'foo',
        link: 'foo',
        text: 'foo'
      }}
      cardBlock2={{
        color: 'foo',
        image: 'foo',
        link: 'foo',
        text: 'foo'
      }}
    />
  );

  expect(component).toMatchSnapshot();
});
