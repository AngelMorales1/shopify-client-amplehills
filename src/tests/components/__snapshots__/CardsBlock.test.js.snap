// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardsBlock
    cardBlock1={
        Object {
            "color": "foo",
            "image": "foo",
            "link": "foo",
            "text": "foo",
          }
    }
    cardBlock2={
        Object {
            "color": "foo",
            "image": "foo",
            "link": "foo",
            "text": "foo",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        null,
      ],
      "className": "w100 p3 mt4 flex content-width mx-auto",
    },
    "ref": null,
    "rendered": Array [
      null,
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          null,
        ],
        "className": "w100 p3 mt4 flex content-width mx-auto",
      },
      "ref": null,
      "rendered": Array [
        null,
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardsBlock
    cardBlock1={Object {}}
    cardBlock2={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        null,
      ],
      "className": "w100 p3 mt4 flex content-width mx-auto",
    },
    "ref": null,
    "rendered": Array [
      null,
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          null,
        ],
        "className": "w100 p3 mt4 flex content-width mx-auto",
      },
      "ref": null,
      "rendered": Array [
        null,
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
